src/PowerShellEditorServices.Protocol/Server/DebugAdapter.cs

```diff
@@ -254,41 +254,38 @@ protected void Stop()
         {
             this.RegisterEventHandlers();
 
-            // Set the working directory for the PowerShell runspace to the cwd passed in via launch.json.
-            string workingDir = launchParams.Cwd;
-
-            // Assuming we have a specified working dir, unescape the path and verify it.
-            if (!string.IsNullOrEmpty(workingDir))
+            // Determine whether or not the working directory should be set in the PowerShellContext.
+            if ((this.editorSession.PowerShellContext.CurrentRunspace.Location == RunspaceLocation.Local) &&
+                !this.editorSession.DebugService.IsDebuggerStopped)
             {
-                workingDir = PowerShellContext.UnescapePath(workingDir);
-                try
+                // Get the working directory that was passed via the debug config
+                // (either via launch.json or generated via no-config debug).
+                string workingDir = launchParams.Cwd;
+
+                // Assuming we have a non-empty/null working dir, unescape the path and verify
+                // the path exists and is a directory.
+                if (!string.IsNullOrEmpty(workingDir))
                 {
-                    if ((File.GetAttributes(workingDir) & FileAttributes.Directory) != FileAttributes.Directory)
+                    workingDir = PowerShellContext.UnescapePath(workingDir);
+                    try
                     {
-                        workingDir = Path.GetDirectoryName(workingDir);
+                        if ((File.GetAttributes(workingDir) & FileAttributes.Directory) != FileAttributes.Directory)
```

Comment:
what is this magical syntax?
`File.GetAttributes(workingDir) & FileAttributes.Directory`

Reply:
Enums as flags. Attributes are usually modelled as flag enums by C# to make them more readable.

Reply:
(https://msdn.microsoft.com/en-us/library/system.io.fileattributes(v=vs.110).aspx)

Reply:
(See [here](https://msdn.microsoft.com/en-us/library/system.io.fileattributes(v=vs.110).aspx))

Reply:
Apparently `HasFlag()` was added in .NET 4.